frontend/actions.js

```diff
@@ -95,23 +95,28 @@ export default ({setState, getState}) => {
     }
   }
 
-  const validateSendAddress = (state) => setState({
-    sendAddress: Object.assign({}, state.sendAddress,
-      {validationError: !Cardano.isValidAddress(state.sendAddress.fieldValue) ? {code: 'SendAddressInvalidAddress'} : ''}),
-  })
+  const validateSendAddress = (state) =>
+    setState({
+      sendAddress: Object.assign({}, state.sendAddress, {
+        validationError: !Cardano.isValidAddress(state.sendAddress.fieldValue)
+          ? {code: 'SendAddressInvalidAddress'}
+          : null,
+      }),
+    })
 
   const validateSendAmount = (state) => {
-    let validationError = ''
-    const amount = parseFloat(state.sendAmount.fieldValue)
-    if (isNaN(amount)) {
+    let validationError = null
+    const strAmount = state.sendAmount.fieldValue
+    const amount = parseFloat(strAmount)
+
+    if (isNaN(amount) || strAmount.toLowerCase().includes('e')) {
       validationError = {code: 'SendAmountIsNan'}
+    } else if (amount <= 0) {
+      validationError = {code: 'SendAmountIsNotPositive'}
+    } else if (strAmount.split('.').length === 2 && strAmount.split('.')[1].length > 6) {
```

Comment:
I guess we should first check precision and only then if the truncated number if positive